pennylane/measurements/probs.py

```diff
@@ -260,6 +260,21 @@ def process_counts(self, counts: dict, wire_order: Wires) -> np.ndarray:
 
         return prob_vector
 
+    def process_density_matrix(self, density_matrix: Sequence[complex], wire_order: Wires):
+        if len(np.shape(density_matrix)) == 2:
+            prob = qml.math.diagonal(qml.math.real(density_matrix))
+        else:
+            prob = qml.math.array(
+                [
+                    qml.math.diagonal(qml.math.real(density_matrix[i]))
+                    for i in range(np.shape(density_matrix)[0])
+                ]
+            )
```

Comment:
There's no need to extract the real part of the diagonal. The density matrix has a complex dtype, but the values along the diagonal should be real. Moreover, `ProbabilityMP.process_state()` extracts the real/imag parts for computing the probability distribution anyway (see [here](https://github.com/PennyLaneAI/pennylane/blob/14a0b63fb50b826c193ad2f59a72397a2463ae8b/pennylane/measurements/probs.py#L211))

```suggestion
        if len(np.shape(density_matrix)) == 2:
            prob = qml.math.diagonal(density_matrix)
        else:
            prob = qml.math.array(
                [
                    qml.math.diagonal(density_matrix[i])
                    for i in range(np.shape(density_matrix)[0])
                ]
            )
```